tests/conftest.py

```diff
@@ -70,3 +95,295 @@ def fin():
+
+
+@pytest.fixture(scope="session")
+def engine():
+    """Test-session-wide database engine"""
+    with testing.postgresql.Postgresql() as pg:
+        engine = create_engine(pg.url())
+        engine.execute("create extension postgis")
+        engine.execute(CreateSchema("crmp"))
+        pycds.Base.metadata.create_all(bind=engine)
+        # sqlalchemy.event.listen(
+        #     pycds.weather_anomaly.Base.metadata,
+        #     'before_create',
+        #     DDL('''
+        #         CREATE OR REPLACE FUNCTION crmp.DaysInMonth(date) RETURNS double precision AS
+        #         $$
+        #             SELECT EXTRACT(DAY FROM CAST(date_trunc('month', $1) + interval '1 month' - interval '1 day'
+        #             as timestamp));
+        #         $$ LANGUAGE sql;
+        #     ''')
+        # )
+        # pycds.weather_anomaly.Base.metadata.create_all(bind=engine)
+        yield engine
+
+
+@pytest.fixture(scope="function")
+def session(engine):
+    """Single-test database session. All session actions are rolled back on teardown"""
+    session = sessionmaker(bind=engine)()
+    # Default search path is `"$user", public`. Need to reset that to search crmp (for our db/orm content) and
+    # public (for postgis functions)
+    session.execute("SET search_path TO crmp, public")
+    # print('\nsearch_path', [r for r in session.execute('SHOW search_path')])
+    yield session
+    session.rollback()
+    session.close()
+
+
+@pytest.fixture(scope="module")
+def mod_blank_postgis_session():
+    with testing.postgresql.Postgresql() as pg:
+        engine = create_engine(pg.url())
+        engine.execute("create extension postgis")
+        engine.execute(CreateSchema("crmp"))
+        sesh = sessionmaker(bind=engine)()
+        yield sesh
+
+
+@pytest.fixture(scope="module")
+def mod_empty_database_session(mod_blank_postgis_session):
+    sesh = mod_blank_postgis_session
+    engine = sesh.get_bind()
+    pycds.Base.metadata.create_all(bind=engine)
+    pycds.weather_anomaly.Base.metadata.create_all(bind=engine)
+    yield sesh
+
+
+@pytest.yield_fixture(scope="function")
+def blank_postgis_session():
+    with testing.postgresql.Postgresql() as pg:
+        engine = create_engine(pg.url())
+        engine.execute("create extension postgis")
+        engine.execute(CreateSchema("crmp"))
+        sesh = sessionmaker(bind=engine)()
+
+        yield sesh
+
+
+@pytest.fixture(scope="function")
+def test_session(blank_postgis_session):
+
+    engine = blank_postgis_session.get_bind()
+    pycds.Base.metadata.create_all(bind=engine)
+
+    # Make sure spatial extensions are loaded for each connection, not just the current session
+    # https://groups.google.com/d/msg/sqlalchemy/eDpJ-yZEnqU/_XJ4Pmd712QJ
+    @event.listens_for(engine, "connect")
+    def connect(dbapi_connection, connection_rec):
+        dbapi_connection.enable_load_extension(True)
+        dbapi_connection.execute("select load_extension('mod_spatialite')")
```

Comment:
If we're just using `testing.postgresql` we shouldn't need the spatialite extension here. That just for using SQLite db backends.